src/client/eppo-client-with-bandits.spec.ts

```diff
@@ -40,16 +47,179 @@ describe('EppoClient Bandits E2E test', () => {
     const configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, banditStore);
     await configurationRequestor.fetchAndStoreConfigurations();
+  });
 
+  beforeEach(() => {
     client = new EppoClient(flagStore, banditStore);
     client.setIsGracefulFailureMode(false);
+    client.setAssignmentLogger({ logAssignment: mockLogAssignment });
+    client.setBanditLogger({ logBanditAction: mockLogBanditAction });
+    jest.clearAllMocks();
   });
 
   afterAll(() => {
     jest.restoreAllMocks();
   });
 
-  it('TODO', () => {
-    // TODO: bandit test cases with the client
+  describe('Shared test cases', () => {
+    const testData = readBanditTestData();
+    // Build a map for more useful test names
+    const testsByFlagKey: Record<string, BanditTestCase> = {};
+    testData.forEach((testCase) => (testsByFlagKey[testCase.flag] = testCase));
+
+    it.each(Object.keys(testsByFlagKey))(
+      'Shared bandit test case - %s',
+      async (flagKey: string) => {
+        const { defaultValue, subjects } = testsByFlagKey[flagKey];
+        let numAssignmentsChecked = 0;
+        subjects.forEach((subject) => {
+          // TODO: handle already-bucketed attributes
+          // TODO: common test case with a numeric value passed as a categorical attribute and vice verse
+
+          const actions: Record<string, Attributes> = {};
+          subject.actions.forEach((action) => {
+            actions[action.actionKey] = {
+              ...action.numericAttributes,
+              ...action.categoricalAttributes,
+            };
+          });
+
+          const subjectAttributes = {
+            ...subject.subjectAttributes.numeric_attributes,
+            ...subject.subjectAttributes.categorical_attributes,
+          };
+
+          const banditAssignment = client.getBanditAction(
+            flagKey,
+            subject.subjectKey,
+            subjectAttributes,
+            actions,
+            defaultValue,
+          );
+
+          // Do this check in addition to assertions to provide helpful information on exactly which
+          // evaluation failed to produce an expected result
+          if (
+            banditAssignment.variation !== subject.assignment.variation ||
+            banditAssignment.action !== subject.assignment.action
+          ) {
+            console.error(
+              `Unexpected result for flag ${flagKey} and subject ${subject.subjectKey}`,
+            );
+          }
+
+          expect(banditAssignment.variation).toBe(subject.assignment.variation);
+          expect(banditAssignment.action).toBe(subject.assignment.action);
+          numAssignmentsChecked += 1;
+        });
+        // Ensure that this test case correctly checked some test assignments
+        expect(numAssignmentsChecked).toBeGreaterThan(0);
+      },
+    );
+
+    //TODO: make this a shared data test case
+    it('Returns default value if no actions provided', () => {
```

Comment:
I _think_ this is what we want (vs. returning the bandit variation and a null action), but we've never formally discussed it.